connection.go

```diff
@@ -129,6 +143,139 @@ func (c *conn) ExecContext(ctx context.Context, query string, args []driver.Name
 	return &res, nil
 }
 
+func Succeeded(response *http.Response) bool {
+	if response.StatusCode == 200 || response.StatusCode == 201 || response.StatusCode == 202 || response.StatusCode == 204 {
+		return true
+	}
+	return false
+}
+
+func (c *conn) HandleStagingPut(presignedUrl string, headers map[string]string, localFile string) (driver.Result, error) {
+	if localFile == "" {
+		return nil, fmt.Errorf("cannot perform PUT without specifying a local_file")
+	}
+	client := &http.Client{}
+	req, _ := http.NewRequest("GET", presignedUrl, nil)
+
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	content, err := io.ReadAll(res.Body)
+
+	if err != nil || !Succeeded(res) {
+		return nil, fmt.Errorf("staging operation over HTTP was unsuccessful: %d-%s", res.StatusCode, content)
+	}
+	return driver.ResultNoRows, nil
+
+}
+
+func (c *conn) HandleStagingGet(presignedUrl string, headers map[string]string, localFile string) (driver.Result, error) {
+	if localFile == "" {
+		return nil, fmt.Errorf("cannot perform GET without specifying a local_file")
+	}
+	client := &http.Client{}
+	req, _ := http.NewRequest("GET", presignedUrl, nil)
+
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	content, err := io.ReadAll(res.Body)
+
+	if err != nil || !Succeeded(res) {
+		return nil, fmt.Errorf("staging operation over HTTP was unsuccessful: %d-%s", res.StatusCode, content)
+	}
+
+	err = os.WriteFile(localFile, content, 0644)
+	if err != nil {
+		return nil, err
+	}
+	return driver.ResultNoRows, nil
+
+}
+
+func (c *conn) HandleStagingDelete(presignedUrl string, headers map[string]string) (driver.Result, error) {
+	client := &http.Client{}
+	req, _ := http.NewRequest("DELETE", presignedUrl, nil)
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+	res, err := client.Do(req)
```

Comment:
are we doing any retries for all the client.do calls?

Reply:
nope, that seems like overkill.

Reply:
I will leave it up to Raymond, if that's a necessary feature we can implement it, but i don't believe we do these retries in go or python

Reply:
Note that in Python we do retry these requests automatically. Presumably that happens automatically in Node as well because we use `axios`. So Jade's feedback makes sense here. Does the Go http library ever automatically retry?

Reply:
Outside of retriable libraries, no. https://github.com/hashicorp/go-retryablehttp -> we could hypothetically use this?

Reply:
At the very least, this seems like follow up PR material given that it is a little more involved. This PR has been sitting a bit, and doesn't really break anything. I think we should prioritize getting it in, unless we see customers complain about retry issues.